refnx/reduce/platypusnexus.py

```diff
@@ -626,6 +627,118 @@ def _check_sample_environments(self, d, h5d):
         return d
 
 
+class SpinChannel(Enum):
+    """
+    Describes the spin state of a polarised neutron beam.
+    """
+
+    UPUP = (1, 1)
+    UPDOWN = (1, 0)
+    DOWNUP = (0, 1)
+    DOWNDOWN = (0, 0)
+
+
+class SpinSet(object):
+    """
+    Describes a set of spin-channels at a given angle.
+
+    TODO: implement polarisation efficiency correction within this class
+
+    Parameters
+    ----------
+    list of {str, h5data}
+        list of str, or list of h5py file handles pointing to
+        a set of polarised neutron beam files
+
+    data_folder: {str, Path}
+        Path to the data folder containing the data to be reduced.
+    """
+
+    def __init__(self, dataset, data_folder=None):
+
+        # Currently only Platypus has polarisation elements
+        self.reflect_klass = PlatypusNexus
+
+        self.data_folder = data_folder
+        if len(dataset) > 4:
```

Comment:
If we change the function signature this won't be required, but it would be important to check that up_up and down_down, etc, represented the correct spin states.